File: app/lib/features/auth/models/verify_reset_otp_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'verify_reset_otp_request.g.dart';

/// Verify an emailed one-time code and request a short-lived reset token
/// (dto-spec §6).
@JsonSerializable(createFactory: false)
class VerifyResetOtpRequest {
  const VerifyResetOtpRequest({required this.email, required this.otp});

  final String email;
  final String otp;

  Map<String, dynamic> toJson() => _$VerifyResetOtpRequestToJson(this);
}

File: src/main/java/com/footverse/auth/dto/VerifyResetOtpRequest.java
package com.footverse.auth.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;

/**
 * Verifies an emailed one-time code and requests a short-lived reset token (dto-spec §6).
 *
 * @param email the account's email the code was issued to
 * @param otp   the six-digit one-time code
 */
public record VerifyResetOtpRequest(
        @NotBlank @Email String email,
        @NotBlank @Pattern(regexp = "^\\d{6}$") String otp) {
}
